### ISSUE TITLE
Low 5px 2D accuracy, but high 3D Transformation accuracy

### ISSUE THREAD

**Issue by jgcbrouns** [microsoft/singleshotpose#93]
Hi everyone,

I created a synthetic dataset of a household product. This was trained for 1400 epochs (not 700)
![7450](https://user-images.githubusercontent.com/8687567/58554311-ec7cbd00-8216-11e9-9d4c-c67ea48bdae0.PNG)
_Note that the image contains examples where the points estimations were good. There are however many pictures where they are not good._

It consists of 7450 images where 80% is used for training. As you can see, the 5px 2D accuracy is low, which is in contrast with my earlier tests (with different models).

What I don't understand @btekin is how you managed to reach such high accuracies on Linemod with only 200 training images. The images in Linemod are of low quality and the objects are not exactly reach in features...
**When I train a model with only 200 trainingimages, I often do not reach higher accuracies than 15%!**. You speak about pretraining for coordinate predictions by setting the confidence to 0, but then the model would never save its weights because accuracy does never go > 0%. Or should I just save the model every 10 epochs or so?

### What could be the cause of this?
1 possible explanation could be that this object's corners are all outside of the object. But technically it should regresses these points estimates as well as coordinates that are on the object itself?

**_Some side notes:_**
- My 3d model is converted from a .stp CAD model
- 3D coordinates of the boundingbox as labeled by my software seems to be correct (tested independently before training by visualizing them)
- Accuracies and losses plotted over time:
![accuracies](https://user-images.githubusercontent.com/8687567/58554791-31edba00-8218-11e9-8335-70654af07ced.PNG)

@btekin as always I look forward to hearing from you!

**Comment by jgcbrouns** [comment]
Note that this 3D model has reflective properties: it's black surface is highly reflective. I use a picture-wall that is filled with random picture of the VOC dataset and a cubemap + shader to generate *random* reflections in my 3D model. This could be of influence to the model not being able to learn properly. However, when I turn reflections off and replace it for matt surface properties, the results are the same.

**Comment by danieldimit** [comment]
Hi @jgcbrouns you know more on this method than me so I can't help you. I'm interested what learning rate you used and what initial weights. It would also be very kind of you if you could provide a link to your dataset and ply object, so I could test with it too.

**Comment by btekin** [comment]
I wonder how you chose your training data. Is it by randomly sampling 80% of the whole dataset? I think it would be important to have a better sampling of the viewpoints so that you have a large coverage of different viewpoints and distances to your object. We used the same training split with the BB8 paper (Rad & Lepetit, ICCV'17) and this was, I think, sampled such that the objects are seen from various different angles (which allows to use a smaller number of training images, but still reasonably high accuracy). They explain their selection of training images in their paper as follows, it might be helpful in your case as well:

> The training images are selected as in [2], such thatrelative  orientation  between them  should  be  larger  than  athreshold.

Visualizations also show that the object does not contain much texture and somewhat symmetrical. Given these I would find the output qualitative results and the overall accuracy reasonable.
